components/nimbus/src/enrollment.rs

```diff
@@ -847,17 +868,16 @@ impl<'a> EnrollmentsEvolver<'a> {
         // experimenter. Once https://github.com/mozilla/experimenter/issues/8661 is fixed, we can remove the calculation
         // for `is_already_enrolled` above, the `put` call here and the `put` method declaration, and replace it with
         // let th = self.targeting_helper;
-        let th = self
-            .targeting_helper
-            .put("is_already_enrolled", is_already_enrolled);
+        let th = self.targeting_helper.lock().unwrap();
+        let targeting_helper = th.put("is_already_enrolled", is_already_enrolled);
```

Comment:
WHY OH WHY does put() clone itself and return a new TargetingHelper instead of mutating itself?

Reply:
🤷